### ISSUE TITLE
Publishing F# function from command line tool doesn't work.

### ISSUE THREAD

**Issue by mariomeyrelles** [Azure/azure-functions-core-tools#567]
Good morning, 

The problem: I have a precompiled F# project. I can't run func azure functionapp publish <<webappname>> on the F# project. It returns the following error running on any folder of the project (root folder, publish folder, bin folder, fsproj folder, etc.)

    Can't determin Project to build. Expected 1 .csproj but found 0

The workaround: I can navigate to inside VS Code:

    \bin\Debug\netcoreapp2.0\publish

Then, I can right-click and select "Deploy to Function App". This is an Azure Functions Extensions to VS Code that works fine. 

![image](https://user-images.githubusercontent.com/839937/42133585-032022e6-7d03-11e8-9d3b-170ae9136854.png)

I'm using Azure Functions V2. My dotnet --info is:

dotnet --info
SDK do .NET Core (refletindo qualquer global.json):
 Version:   2.1.300
 Commit:    adab45bf0c

Ambiente de tempo de execução:
 OS Name:     Windows
 OS Version:  10.0.16299
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.1.300\

Host (useful for support):
  Version: 2.1.0
  Commit:  caa7b7e2ba

.NET Core SDKs installed:
  1.0.0-rc4-004771 [C:\Program Files\dotnet\sdk]
  1.0.0 [C:\Program Files\dotnet\sdk]
  1.0.1 [C:\Program Files\dotnet\sdk]
  1.1.0 [C:\Program Files\dotnet\sdk]
  2.1.4 [C:\Program Files\dotnet\sdk]
  2.1.104 [C:\Program Files\dotnet\sdk]
  2.1.201 [C:\Program Files\dotnet\sdk]
  2.1.300-preview1-008174 [C:\Program Files\dotnet\sdk]
  2.1.300 [C:\Program Files\dotnet\sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.1.0-preview1-final [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.All 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.1.0-preview1-final [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 1.0.3 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.4 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.1 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 1.1.2 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.5 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.6 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.0.7 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0-preview1-26216-03 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.0 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download

**Comment by mariomeyrelles** [comment]
The issue seems to be related to this code: 

https://github.com/Azure/azure-functions-core-tools/blob/ed13074afceb712d6f42594292a4d3c3297251ec/src/Azure.Functions.Cli/Helpers/DotnetHelpers.cs#L75

**Comment by ahmelsayed** [comment]
@mariomeyrelles The reason for this error is that the cli now does csproj by default which requires a build step before publishing. 

I'm assuming in your case you have an fsproj? When you run publish, do you expect it to build your fsproj, it never did that, so I'm assuming you have another publish flow? like build then run `func publish` from the output dir?

**Comment by mariomeyrelles** [comment]
Hello @ahmelsayed ,

Exactly. I have another build flow, usually with Paket and Fake (F# tools that people usually consider when they develop in F#). I publish to bin folder using `dotnet publish`. Then I ocasionally deploy to Azure.

My workflow is heavily inspired by this example: [https://github.com/mikhailshilkov/azure-functions-fsharp-examples#5-precompiled-function](https://github.com/mikhailshilkov/azure-functions-fsharp-examples#5-precompiled-function) and also, this example: [https://github.com/mikhailshilkov/azure-functions-fsharp-examples/tree/master/8-paket](https://github.com/mikhailshilkov/azure-functions-fsharp-examples/tree/master/8-paket).

As a side note, I noticed that when you register an extension (add a nuget package) using a command like `func extension install xxxxxxxx`, it creates a csproj containing the package reference. Maybe this is the rationale behind the need to use a .csproj file to build something before publishing to Azure.

**Comment by mariomeyrelles** [comment]
Thanks MaxWilson... I'll give this a try.

**Comment by mixtura** [comment]
@mariomeyrelles Did you manage to solve this problem? The solution provided by @MaxWilson didn't work for me.

**Comment by mariomeyrelles** [comment]
@mixtura Nope. I'm deploying using the menu option "Deploy to Function App..." as shown in the image. I'm ok with this option for now...
